src/plot.js

```diff
@@ -97,19 +99,20 @@ export function plot(options = {}) {
         ? facetsExclude || (facetsExclude = facetsIndex.map((f) => Uint32Array.from(difference(facetIndex, f))))
         : undefined;
     const {data, facets, channels} = mark.initialize(markFacets, facetChannels);
+    applyScaleTransforms(channels, options);
+    stateByMark.set(mark, {data, facets, channels});
+
+    // Warn for the common pitfall of wanting to facet mapped data.
     if (
-      facets && // faceting is present
-      facetIndex.length > 1 && // more than one facet (non-trivial faceting)
+      facetIndex?.length > 1 && // non-trivial faceting
       mark.facet === "auto" && // no explicit mark facet option
-      mark.data !== facet.data && // mark is implicitly not faceted (different data)
-      arrayify(mark.data)?.length === facetIndex.length // mark data seems parallel to facet data
+      mark.data !== facet.data && // mark not implicitly faceted (different data)
+      (data && data.length) === facetDataLength // mark data seems parallel to facet data
```

Comment:
This is now testing the cardinality of the possibly-transformed data, rather than the original mark data. I think we should be looking at the original mark data as I had in my suggestion (since we’re using that same data for the equality test with “auto” facet). Why switch to the transformed data here?

Reply:
Sorry, my bad! Will fix in #1014